### PR TITLE
Fix/tao 10137/fix issues when delivery created event triggered without test

### PR DESCRIPTION
**Pull request by SashaTAO**
JIRA ticket: https://oat-sa.atlassian.net/browse/TAO-10137

Only URI property of `originTest` in DeliveryCreatedEvent is used in event handlers and is serialized. Additionally when DeliveryCreatedEvent is triggered client code doesn't always have test resource or event test uri. Therefore:
- constructor parameter was changed to `string $originTestUri`
- added method to get test uri. If it was not passed as constructor parameter try  to  read delivery property
- if delivery doesn't have  origin  assembly test property return `null` as default value (this case must be properly handled by client code)

### CODE REVIEW

File: model/event/DeliveryCreatedEvent.php [hunk 1]
```diff
@@ -82,19 +84,27 @@ public function getWebhookEventName()
...
     }
 
-    public function getOriginTest(): ?core_kernel_classes_Resource
+    /**
+     * @return string|null
+     */
+    public function getOriginTestUri(): ?string
     {
-        return $this->originTest;
+        if ($this->originTestUri === null) {
```
Comment (wazelin) [inline model/event/DeliveryCreatedEvent.php:101]: Why is there a need to have that property's value defined externally via the constructor then?
Do we want it to be overridable?

Comment (SashaTAO) [reply model/event/DeliveryCreatedEvent.php:101]: This logic was already [there](https://github.com/oat-sa/extension-tao-delivery-rdf/pull/345/files#diff-236c31dbbffa3833f834b0a3eedca3dcL92). In this  PR I extracted it to a separate method to be able to  reuse and left this logic here for backward compatibility to cover cases when DeliveryCreatedEvent is still triggered without test URI. Would you suggest to remove it and return `null` if test URI was not passed in constructor?

Comment (wazelin) [reply model/event/DeliveryCreatedEvent.php:101]: No-no, I mean, why do we even need to pass it to constructor then?

Comment (SashaTAO) [reply model/event/DeliveryCreatedEvent.php:101]: To avoid unnecessary DB call in case it was passed as constructor parameter.

**Pull request closed without merging**